### PR TITLE
chore(maintenance): migrate metrics utility to biome

### LINKED ISSUE #2799: Maintenance: add biome to project (metrics)
### Summary

With #2791 being merged, we can now migrate the `packages/metrics` workspace to use Biome.

As part of the migration I will fix any issue/inconsistency that arises. I'll do my best to minimize changes and leave the logic untouched.

### Why is this needed?

So that we can gradually replace `eslint`, `prettier`, and all their dependencies with a single one, `@biomejs/biome`.

### Which area does this relate to?

Other

### Solution

_No response_

### Acknowledgment

- [X] This request meets [Powertools for AWS Lambda (TypeScript) Tenets](https://docs.powertools.aws.dev/lambda/typescript/latest/#tenets)
- [ ] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Python](https://github.com/aws-powertools/powertools-lambda-python/), [Java](https://github.com/aws-powertools/powertools-lambda-java/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### Future readers

Please react with 👍 and your use case to help us understand customer demand.

### PR DESCRIPTION
**Pull request by daschaa**
### Changes

Closes #2799 

**Issue number: closes #2799** 

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Comment by dreamorosi** [comment]
> No related issues found. Please ensure there is an open issue related to this change to avoid significant delays or closure.

Sorry about this @daschaa - we need to improve our automation (I'm working on that on the side) to better detect linked issues.

In the meantime, as a workaround - if you want to avoid the bot nagging you - you could edit/write the PR body with the issue number like [here](https://github.com/aws-powertools/powertools-lambda-typescript/pull/2814#issue-2425172941) (I edited your comment to fix it) 😁

**Comment by daschaa** [comment]
@dreamorosi I think with this regex the automation might work (at least i get good matches):

```regex
\b\*{0,2}[iI]ssue [nN]umber\*{0,2}:\*{0,2}\s*(?<closingWord>closes?|closed|fix|fixes?|fixed|resolves?|resolved)*?\s*#*(?<issue>\d+)\b
```

File: packages/metrics/tests/e2e/basicFeatures.decorators.test.ts [hunk 1]
```diff
@@ -1,29 +1,24 @@
-/**
```
Comment (dreamorosi) [inline packages/metrics/tests/e2e/basicFeatures.decorators.test.ts:1]: I think this was removed by mistake

**Review by dreamorosi — COMMENTED** [review]
Thank you for the work on this.

I have left only one other comment, but seems we are going in the right direction.

We'll have to do something about all the `// biome-ignore  lint/complexity/useLiteralKeys` but since it'll likely require a refactor of either the tests or the utility we will address this separately (same for Logger).

**Comment by dreamorosi** [comment]
@daschaa - I just saw there are [two Sonar findings](https://sonarcloud.io/project/issues?id=aws-powertools_powertools-lambda-typescript&pullRequest=2816&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true).

They are not areas you've touched, but the way the tool works is that it runs on files that were edited as part of the PR, and scans them. So that's why they're coming up now.

**Comment by daschaa** [comment]
> @daschaa - I just saw there are [two Sonar findings](https://sonarcloud.io/project/issues?id=aws-powertools_powertools-lambda-typescript&pullRequest=2816&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true).

I'll check them :) Sorry that I did not catch them earlier. I guess the checkmark icon in the sonar report just let's my brain ignore the report. 😄

**Review by dreamorosi — APPROVED** [review]
Thank you so much for the work on this PR!

We're going to merge this for now, despite the addition of `lint/complexity/useLiteralKeys` in the unit tests. We have another conversation ongoing in #2821 to make changes that we hope will allow us to remove most/all of the suppressions (thank you for opening the issue 😉).

**Pull request merged**